File: palindrome_number/palindrome_number.c
bool isPalindrome(int x) {
    if (x == 0){
        return true;
    }
    
    if (x<0 || x%10 == 0){
        return false;
    }
    
    int reversed_number = 0;
    while (x>reversed_number){
        reversed_number = reversed_number * 10 + x%10;
        x = floor(x/10);
    }
    if ( x == reversed_number || floor(reversed_number/10)==x){
        return true;
    }
    else{
        return false;
    }
}

File: palindrome_number/palindrome_number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:  
            return False  
        numbers_list = list(str(x))
        numbers_lists = list(map(int, numbers_list))
        for i in range(int(len(numbers_list)/2)):
            if numbers_list[i] == numbers_list[-i -1]:  
                continue
            else:
                return False
        return True
